### PR TITLE
Use env for revision id if not passed in as param; use `git describe` as backup

### PR DESCRIPTION
**Pull request by samnoyes**

### CODE REVIEW

File: libs/langchain/langchain/smith/evaluation/runner_utils.py [hunk 1]
```diff
@@ -1330,6 +1335,24 @@ def run_on_dataset(
     return container.finish(batch_results, verbose=verbose)
 
 
+def _get_default_revision_id() -> Optional[str]:
```
Comment (eyurtsev) [inline libs/langchain/langchain/smith/evaluation/runner_utils.py:1338]: Could you add a description of what `LANGCHAIN_REVISION_ID` is? And which `git` repo is this for?

Comment (eyurtsev) [reply libs/langchain/langchain/smith/evaluation/runner_utils.py]: Do we need to memoize the result if `git describe --tags --dirty` fails? e.g., no git installed or no .git found?

**Review by eyurtsev — APPROVED** [review]
This works

File: libs/langchain/langchain/smith/evaluation/runner_utils.py (hunk 1 above)
Comment (hinthornw) [reply libs/langchain/langchain/smith/evaluation/runner_utils.py]: Ya - he's implementing in the SDK also! we memoize there

**Pull request merged**